### ISSUE TITLE
Provide more compact metadata annotation format

### ISSUE THREAD

**Issue by anderseknert** [open-policy-agent/opa#5402]
When working with many annotated packages and rules, the current "# METADATA" format for annotations feel rather verbose, and the fact that nothing else in Rego is written in capital letters makes them stand out more than they ought to. Additionally, the "# METADATA" part itself occupies a line of code that could be used for something better. Providing a more compact alternative would be nice. I suggest using `##` at the beginning of a line to mean this comment is part of a metadata annotation.

**old**
```rego
# METADATA
# title: foo
# description: bar
allow if ...
```
**new*
```rego
## title: foo
## description: bar
allow if ...
```

This would also allow mixing regular comments with metadata ones all in the same block, e.g:

```rego
## title: foo
#  description below is wrong, but remains for historical reasons
## description: bar
allow if ...
```

**Comment by tsandall** [comment]
This seems nice to have. I'd opt for a more explicit syntax, e.g., instead of `##` use `#@` or something that stands out a bit more. I know that I'll often comment out several lines while editing and sometimes that might include comments because it's just convenient. Some editors will inject a space after the comment character, others won't. I would not want that to accidentally generate a bunch of annotations.. For example:

```rego
# some comment ...
allow { input.foo == "bar" }

# some other comment ...
allow { input.foo == "baz" }
```

Could become:

```rego
## some comment ...
#allow { input.foo == "bar" }
#
## some other comment ...
#allow { input.foo == "baz" }
```

**Comment by anderseknert** [comment]
Fair point about editors. I guess a simple way to avoid that would be to say that `##` _and_ a valid metadata attribute starts a metadata block, so this would be considered one:

```rego
## title: foo
```

but not 

```rego
##allow { input.foo == "baz" }

# or

## allow { input.foo == "baz" }
```

But `#@` is good in that it clearly differentiates metadata comments from normal comments, so let's do that then.

```rego
#@ title: foo
#@ description: bar
```

**Comment by anderseknert** [comment]
Although that's an interesting idea, I think that's one step too far 😅 I think it's good to keep metadata as comments, as it feels like they logically belong in that group, should be syntax highlighted as such, etc.
